package/common_init/overlay/etc/init.d/S99reset_boot_counter

```diff
@@ -0,0 +1,33 @@
+#!bin/sh
+#
+# reset boot counters
+#
+
+export name="reset_boot_counter"
+
+export log_tag=$name
+source /etc/init.d/logging.sh
+
+source /etc/init.d/common.sh
+
+upgrade_uboot_environment()
+{
+  fw_setenv pre_boot_counter 0
+  fw_setenv post_boot_counter 0
+  fw_setenv increment_boot_counter "setexpr pre_boot_counter \${pre_boot_counter} + 1; saveenv"
```

Comment:
this doesn't actually work with a non-upgraded dev image, so you get the follow error:
`Unexpected boot counter values! pre_boot_counter: 0 post_boot_counter 1`

Reply:
This is expected behaviour: this pull request depends on the `setexpr` command being enabled by the u-boot PR.

In the `feature, not a bug` vein, this does serve as an indication that the dev image is "out of date"

Reply:
yeah I would say maybe a warning in that case to upgrade dev image instead of 'unexpected' would be helpful since 0 preboot should be impossible